### DISCUSSION TITLE
YOLOv8 Augmentations

### DISCUSSION THREAD

**Discussion by coopercox315** [ultralytics/ultralytics#14375] [Q&A] [answered]
Hey y'all, I am currently working with a custom dataset I have exported from Roboflow with 5 different augmentations applied and have been experimenting with my model like this for a couple days. I have recently learnt that YOLOv8 actually applies its own augmentations, my question being whether this will interfere with the augmentations I have already applied and possibly make my model worse as I suspect this could mean augmentations are being applied to images which have already being augmented.
Thanks in advance!

**Comment by glenn-jocher** [comment, marked as answer]
@mysticrain32 hello!

Great question! When working with YOLOv8, it's important to understand how augmentations are applied to ensure optimal model performance. YOLOv8 indeed applies its own set of augmentations during training, which can potentially overlap with the augmentations you've already applied using Roboflow.

To avoid redundant augmentations, you have a couple of options:

1. **Disable YOLOv8 Augmentations**: You can disable or customize the augmentations in YOLOv8 by modifying the dataset configuration file (`.yaml`). This way, you can ensure that only the augmentations from Roboflow are applied.

2. **Fine-Tune Augmentation Parameters**: If you prefer to use both sets of augmentations, you can fine-tune the augmentation parameters in YOLOv8 to complement the augmentations from Roboflow. This can help in creating a more diverse dataset without over-augmenting.

Here’s an example of how you can disable or adjust augmentations in the `.yaml` file:

```yaml
augment:
  flipud: 0.0  # Disable vertical flip
  fliplr: 0.5  # Horizontal flip with 50% probability
  mosaic: 0.0  # Disable mosaic augmentation
  # Add or adjust other augmentations as needed
```

For more detailed guidance on data preprocessing and augmentation techniques, you can refer to our [Data Preprocessing Guide](https://docs.ultralytics.com/guides/preprocessing_annotated_data/).

Feel free to experiment with these settings to find the best combination for your dataset. Happy training! 🚀

**Reply by coopercox315** [reply]
thanks so much for the insights @glenn-jocher, that makes a lot of sense and I will play around with that now.
Keep up the great work, y'all have an amazing model and experimenting with it has been amazing!

**Reply by glenn-jocher** [reply]
@mysticrain32 thank you for your kind words! We're glad to hear you're enjoying working with YOLOv8. If you have any further questions or run into any issues, feel free to reach out. Happy experimenting!

**Comment by nachouve** [comment]
Is there a way to generate a series of examples of yolo augmentations performed on some specific images with my .yaml params?

I know that some examples appear in the "runs" folder when training. But I would like to control over which images are made and check the result of the augmentations.

**Reply by pderrenger** [reply]
@nachouve yes, you can generate examples of YOLO augmentations on specific images by using the `augment` argument in the `predict` mode. This allows you to visualize the augmentations applied to your images based on your .yaml parameters. You can specify the images you want to augment and inspect the results directly. This approach provides more control over the images and augmentations, helping you verify the effects before training. For more detailed guidance, please refer to the Ultralytics documentation.
